Assets/UIElements/_Elements/XRUI_Buttons/Scripts/XRUI_2DButtonText.cs

```diff
@@ -26,7 +28,23 @@ protected override void Awake()
         xrFeedback.onTouchEnter.AddListener((XRController controller) => { onClickDown?.Invoke(); });
         xrFeedback.onTouchExit.AddListener((XRController controller) => { onClickUp?.Invoke(); });
     }
-
+    private void Start()
+    {
+     //its here because, just to text awake happens before create interaction on feedback
+        if (xrFeedback.allowDistanceEvents)
+        {
+            XRBaseInteractable interactable = gameObject.GetComponent<XRBaseInteractable>();
+            if (interactable == null)
+                interactable = gameObject.GetComponentInChildren<XRBaseInteractable>();
+
+            if (interactable != null)
+                interactable.onSelectEnter.AddListener((XRBaseInteractor) => { onClickDown?.Invoke(); });
+        }
+        else if (distanceCollider != null)
+        {
+            distanceCollider.SetActive(false);
+        }
+    }
```

Comment:
Same as the last commnet, maybe we can do this on BaseClass or a abstract/virtual function

Reply:
Changed